postgresql-ballerina/client.bal

```diff
@@ -0,0 +1,264 @@
+    #            remaining commands in the batch after a failure. The summary of the executed queries in case of error
+    #            can be accessed as `(<sql:BatchExecuteError> result).detail()?.executionResults`.
+    remote isolated function batchExecute(@untainted sql:ParameterizedQuery[] sqlQueries) returns sql:ExecutionResult[]|sql:Error {
+        if (sqlQueries.length() == 0) {
+            return error sql:ApplicationError("Parameter 'sqlQueries' cannot be empty array");
+        }
+        if (self.clientActive) {
+            return nativeBatchExecute(self, sqlQueries);
+        } else {
+            return error sql:ApplicationError("PostgreSQL Client is already closed, hence further operations are not allowed");
+        }
+    }
+
+    # Executes a SQL stored procedure and returns the result as stream and execution summary.
+    #
+    # + sqlQuery - The query to execute the SQL stored procedure
+    # + rowTypes - The array of `typedesc` of the records that should be returned as a result. If this is not provided
+    #               the default column names of the query result set be used for the record attributes.
+    # + return - Summary of the execution is returned in `ProcedureCallResult` or `sql:Error`
+    remote isolated function call(@untainted string|sql:ParameterizedCallQuery sqlQuery, typedesc<record {}>[] rowTypes = [])
+    returns sql:ProcedureCallResult|sql:Error {
+        if (self.clientActive) {
+            return nativeCall(self, sqlQuery, rowTypes);
+        } else {
+            return error sql:ApplicationError("PostgreSQL Client is already closed, hence further operations are not allowed");
+        }
+    }
+
+    # Close the SQL client.
+    #
+    # + return - Possible error during closing the client
+    public isolated function close() returns sql:Error? {
+        self.clientActive = false;
+        return close(self);
+    }
+}
+
+# Provides a set of configurations for the postgresql client to be passed internally within the module.
+#
+# + host - URL of the database to connect
+# + port - Port of the database to connect
+# + user - Username for the database connection
+# + password - Password for the database connection
+# + database - Name of the database
+# + options - Postgresql datasource `Options` to be configured
+# + connectionPool - Properties for the connection pool configuration. Refer `sql:ConnectionPool` for more details
+
+type ClientConfiguration record {|
+    string host;
+    int port;
+    string? user;
+    string? password;
+    string? database;
+    Options? options;
+    sql:ConnectionPool? connectionPool;
+|};
+
+# PostgreSQL database options.
+#
+# + ssl - SSL Configuration to be used
+# + connectTimeout - Timeout to be used when connecting to the postgresql server.
+# + socketTimeout - Socket timeout during the read/write operations with postgresql server,
+#                            0 means no socket timeout.
```

Comment:
`0 means no socket timeout.`

is this the way normally do for other postgres clients? since this is an optional field, if the user not provided, we consider it as no socket timeout right?

Reply:
what about other timeouts configs?

Reply:
`0 means no socket timeout` is a driver level condition.
If the user not provided, we do not add that parameters into the datasource. So default values will be use by the driver.

Reply:
Timeout parameters were updated. Commit id is 79ccf2074765e4298ea0788e22a3bc3e33604b15.